### PR TITLE
go: migrate CI workflow into repo-templates

### PR DESCRIPTION
**Pull request by bgilbert**

### CODE REVIEW

File: go/tests.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    - name: Run linter
+      uses: golangci/golangci-lint-action@v3
+{%- if do_multi_os %}
+      if: runner.os == 'Linux'
+{%- endif %}
+      with:
+        version: {{ golangci_lint_version }}
+        args: -E=gofmt --timeout=30m0s
+
+{%- if git_repo | default(value="") == "ignition" %}
```
Comment (jlebon) [inline go/tests.yml:82]: Hmm, seems weird to have a template section specific to one repo in this repo. But I understand wanting to centralize the Go matrix. Would it be cleaner to have this bit live in a separate file?

**Review by jlebon — APPROVED** [review]
Looks sane overall!

**Comment by bgilbert** [comment]
Updated!

**Review by dustymabe — APPROVED** [review]
I'm not familiar with all the pieces but the strategy LGTM. Thanks for working on it!

**Pull request merged**